### DISCUSSION TITLE
Improvements to Table of Contents

### DISCUSSION THREAD

**Discussion by Loriel-S-Parloa** [GitbookIO/community#360] [Feature Requests]
I would really like to be able to add a table of contents on certain pages. 
Right now, I have have to create a bulleted list and link each individually. It's time consuming, and also not so visually appealing.
Thank you!

**Comment by rodrcastro** [comment]
Hey @Loriel-S-Parloa! Thank you for your feedback! GitBook already has a table of contents by default, both on the page and the space. 

The table of contents is what you see on the left of a space, with the pages you have inside that space. [You can read more about it our documentation here.](https://docs.gitbook.com/content-creation/content-structure/content-in-a-space#table-of-contents)

As for the content on the page itself, we have a section called "On this page" on the right-hand side of each page. [We also have more information on it in our documentation](https://docs.gitbook.com/product-tour/navigation#on-this-page), but whenever you add an H1 or H2 to your page, we automatically generate and update that list with the changes you make. It links to each section of your page underneath those headings.

Hopefully, those are what you are looking for ;)

**Reply by victor-chebotar** [reply]
The table of contents on the page is visible only on the big screen. 

I'd **prefer to hide space content** over page content. Because when I work with pages I need page content first.

(I use only https://app.gitbook.com/ UI)

excerpt from Gitbook docs:
> If you can't see the right-hand column of the app, this may be because your browser window is less than 1430 pixels wide. Your browser window will need to be at least 1430 pixels wide to see and use the on this page section.

**Comment by LerMerGerd** [comment]
It would be very useful to be able to add a block to a page (within an Expandable block too) which would dynamically display a table of contents containing the Child pages of the Parent page that I select. Maybe it's called something else since GitBook does already have a "Table of Contents"

This would be helpful as I have some documentation that references other articles, and sometimes I want to list all of those child articles without having to manually create my own links (like the OP mentioned). Creating my own links means that I would have to remember to manually create new links for any new pages within that Parent page, or remove links that might be outdated.

**Reply by LerMerGerd** [reply]
I'd like to add that my note is regarding a TOC of the child pages under a specific parent page - but this is in addition to the OP's request of placing a TOC as a block on the page, showing all of the headers I have on the page - not just H1 and H2 like the On This Page tool does. And to be clear, I don't think the On This Page tool should go past the H2.

It is helpful to clients to have the full TOC listed so they can click directly to the spot of the article that they know they need to look at. Here is an example article where we had to create our own TOC - https://help.checkoutchamp.com/funnel-builder/onboarding-and-activation

**Comment by MikkoSairo** [comment]
It appears this has been requested a while ago, but I would like to make the same or similar request as already suggested here.

I would like to be able to add a Space-level or Pages under current page -level ToC as a block to my content. 

It would help with space/group level landing pages, which can be quite empty or require manual upkeep for mobile or small screen users. Without ToC or other selected links, the navigation experience for those users feels a bit lacking. 

For mobile/small screen users, the top-level section/space nav stays visible, while the current space's side navigation is hidden in hamburger menu which is above the section navigation. If the landing page has no navigational features or other content, it appears like there's no content at all in that space.
